packages/compass-e2e-tests/helpers/compass.ts

```diff
@@ -787,7 +805,45 @@ export async function startBrowser(
     ...webdriverOptions,
     ...wdioOptions,
   })) as CompassBrowser;
-  await browser.navigateTo('http://localhost:7777/');
+
+  if (TEST_ATLAS_CLOUD_EXTERNAL_URL) {
+    // Navigate to a 404 page to set cookies
+    await browser.navigateTo(`https://${TEST_ATLAS_CLOUD_EXTERNAL_URL}/404`);
+
+    const cookiesFile = process.env.COOKIES_FILE;
+    if (!cookiesFile) {
+      throw new Error(
+        'ATLAS_DOMAIN is set but COOKIES_FILE is not. Please set COOKIES_FILE to the path of the cookies file.'
+      );
+    }
+    const cookies: StoredAtlasCloudCookies = JSON.parse(
+      await fs.readFile(cookiesFile, 'utf8')
+    );
+
+    for (const cookie of cookies) {
+      if (
+        cookie.name.includes('mmsa-') ||
+        cookie.name.includes('mdb-sat') ||
+        cookie.name.includes('mdb-srt')
+      ) {
+        await browser.setCookies({
+          name: cookie.name,
+          value: cookie.value,
+          domain: cookie.domain,
+          path: cookie.path,
+          secure: cookie.secure,
+          httpOnly: cookie.httpOnly,
+        });
```

Comment:
`setCookies` [takes an array](https://v8.webdriver.io/docs/api/browser/setCookies#parameters), you don't need to do this in sequence